File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        maxDiff = -1
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if nums[j]>nums[i]:
                    maxDiff = max(maxDiff,(nums[j]-nums[i]))

        return maxDiff

File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.rs
impl Solution {
    pub fn maximum_difference(nums: Vec<i32>) -> i32 {
        let mut max_diff: i32 = -1;
        let n = nums.len();

        for i in 0..n {
            for j in (i + 1)..n {
                if nums[j] > nums[i] {
                    max_diff = max_diff.max(nums[j] - nums[i]);
                }
            }
        }

        max_diff
    }
}